### PR TITLE
Minor clean up work renaming of classes and updating/adding JavaDoc

### PR DESCRIPTION
**Pull request by dpwspoon**
I was trying to find the CLI and it took me a little bit to end up at the CommandProcessors so I renamed it.  If your in opinion that it shouldn't change I'm alright with that.

Changed Gateway Listener to Observer, as I felt it more accurately reflected observing changes in the gateway.  Added Javadoc to the Observer

### CODE REVIEW

File: server/src/main/java/org/kaazing/gateway/server/GatewayCLI.java [hunk 1]
```diff
@@ -32,20 +32,18 @@
...
- * value of <configFile> is considered as a path to the gateway configuration file--NOT just a file name. If provided by the
- * caller, the property is converted to a system property and passed to Gateway, rather than requiring Gateway to handle a new
- * input vector.
+ * Supported command line arguments are --config <configFile> and --help. The value of <configFile> is considered as a
+ * path to the gateway configuration file--NOT just a file name. If provided by the caller, the property is converted to
+ * a system property and passed to Gateway, rather than requiring Gateway to handle a new input vector.
  */
-public class GatewayCommandProcessor {
-    private static final Logger LOGGER = LoggerFactory.getLogger(GatewayCommandProcessor.class);
+public class GatewayCLI {
```
Comment (krismcqueen) [inline server/src/main/java/org/kaazing/gateway/server/GatewayCLI.java]: Could this be GatewayCommandLineProcessor or something similar that isn't an abbreviation?  Everything else looks good.

File: server/src/main/java/org/kaazing/gateway/server/WindowsMain.java [hunk 2]
```diff
@@ -64,7 +61,7 @@ public static void main(String... args) throws Exception {
             args[i] = arg;
         }
 
-        GatewayCommandProcessor commandProcessor = new GatewayCommandProcessor(formatter);
+        GatewayCLI commandProcessor = new GatewayCLI(formatter);
```
Comment (krismcqueen) [inline server/src/main/java/org/kaazing/gateway/server/WindowsMain.java]: maybe rename the variable too based on the final class name

File: server/src/main/java/org/kaazing/gateway/server/GatewayCLI.java (hunk 1 above)
Comment (dpwspoon) [inline server/src/main/java/org/kaazing/gateway/server/GatewayCLI.java]: Sounds good,  Done!

File: server/src/main/java/org/kaazing/gateway/server/WindowsMain.java (hunk 2 above)
Comment (dpwspoon) [inline server/src/main/java/org/kaazing/gateway/server/WindowsMain.java]: Sounds good,  Done!

**Comment by dpwspoon** [comment]
Thanks @krismcqueen

**Pull request merged**